rust/src/base.rs

```diff
@@ -539,6 +552,64 @@ impl FromStr for ConsignmentEndpoint {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, StrictEncode, StrictDecode)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate")
+)]
+#[non_exhaustive]
+pub enum InvoiceChain {
+    /// Bitcoin mainnet
+    #[cfg_attr(feature = "serde", serde(rename = "mainnet"))]
+    Mainnet,
+
+    /// Bitcoin testnet version 3
+    #[cfg_attr(feature = "serde", serde(rename = "testnet"))]
+    Testnet3,
+
+    /// Default bitcoin regtest network
+    #[cfg_attr(feature = "serde", serde(rename = "regtest"))]
+    Regtest,
```

Comment:
There is no such thing as regtest. Regtests are multiple, always private setup.

Reply:
What I meant is that we shouldn't have this option here. You can't send a transaction to "a regtest" network without providing much more information (genesis, bitcoin core node address etc)

Reply:
Initially I was referring to bitcoin core default regtest, but there is no meaning in restricting invoices to just a regtest since that network will be used only for test purposes. I don't think there are use cases for regtest invoices outside test purposes so IMO a general `Regtest` enum is more than enough here, we don't need to differentiate different regtest networks in this context but it's still useful to have one

Reply:
This doesn't make sense to me since the user might use different bitcoin core instance - with different regtest - than the invoice creator. In such cases it would be a situation which is really hard to debug. So I think we should have some regtest id here - maybe a u32-fingerprint of a specific regtest genesis or something

Reply:
Invoices are meant to improve the UX of end users to detect mistakes before a state transition is attempted, they are not supposed to solve all the problems that developers testing in regtest may encounter. The worst case scenario here is that the invoice doesn't help the sender to understand he is on the wrong chain, and the send will fail, just like it happens right now with no invoices being used. The regtest id to me seems to add complexity to the invoice format with the scope of benefiting marginally some rare instances where developers are testing using multiple regtests.